### PR TITLE
http_jmap.c: initial implementation of proxying JMAP (based on authenticated user)

### PR DESCRIPTION
**Pull request by ksmurchison**
Each JMAP method should probably be proxied by accountId, but this is a start.

### CODE REVIEW

**Review by rsto — COMMENTED** [review]
The code changes look sane to me. Two questions, though:

- Will this work with web sockets, too?
- Has this been tested? I guess we don't have automated tests for proxying JMAP, so I assume you did manual testing?

**Comment by ksmurchison** [comment]
This currently will not work with WebSockets.
I only tested it by hand.

**Comment by rsto** [comment]
OK, I can't tell if not supporting WebSockets is good enough, but the code at hand looks sane. I'll approve.

**Review by rsto — APPROVED** [review]
Looks good to me.

**Comment by ksmurchison** [comment]
Updated to proxy JMAP/WS over HTTP/1.1.  JMAP/WS over HTTP/2 still to be done

**Comment by elliefm** [comment]
I'd like to see at least a basic cassandane test for this.  We already have the Murder test suite which knows how to set up a frontend/mupdate server and a couple of backends, so the infrastructure work is already done.

A copy of Murder.frontend_commands that uses JMAP instead of IMAP would be a great start, just so we can establish the baseline of "stuff is proxied to the right place, responses come back, actions take effect, and we'll notice if it breaks later"

File: imap/http_jmap.c [hunk 1]
```diff
@@ -327,6 +329,29 @@ static int jmap_parse_path(struct transaction_t *txn)
...
 
+    if (config_mupdate_server && !config_getstring(IMAPOPT_PROXYSERVERS)) {
+        /* Locate the authenticated user's INBOX */
+        char *inbox = mboxname_user_mbox(httpd_userid, NULL);
+        int r = proxy_mlookup(inbox, &tgt->mbentry, NULL, NULL);
+
+        free(inbox);
+
+        if (r) {
+            syslog(LOG_ERR, "mlookup(%s) failed: %s", inbox, error_message(r));
```
Comment (rsto) [inline imap/http_jmap.c:340]: could use xsyslog here

**Review by rsto — CHANGES_REQUESTED** [review]
The code looks good me, except for using syslog. I agree with @elliefm that we it needs at least a basic Cassandane test. Also, I think this feature warrants an entry in the changes directory.

**Pull request closed without merging**